core/src/jsMain/kotlin/SharedRepeatableTask.kt

```diff
@@ -0,0 +1,42 @@
+package com.juul.kable
+
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.CoroutineStart
+import kotlinx.coroutines.CoroutineStart.DEFAULT
+import kotlinx.coroutines.Deferred
+import kotlinx.coroutines.async
+
+/**
+ * A mechanism for launching and awaiting some shared job repeatedly.
+ *
+ * The job is launched by calling [launchAsync]. Subsequent calls to [launchAsync] will return the
+ * same (i.e. shared) job if it is still executing.
+ */
+internal class SharedRepeatableTask(
```

Comment:
Love this. Could it be used to improve the other platform implementations too? Obviously not gonna block a PR on that improvement.

Reply:
Quite possibly as there are variations on this pattern in the other target implementations.